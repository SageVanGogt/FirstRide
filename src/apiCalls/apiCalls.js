import { geoKey } from './../apiKey';

const signinUser = async (user) => {
  const url = 'http://localhost:3000/api/users/';
  const init = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, init);
  const profile = await response.json();
  return profile;
}; 

const signupUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const body = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const userRecieved = await response.json();
  return userRecieved;
};

const fetchDestination = async (name) => {
  const url = `http://localhost:3000/api/locations/${name}`;
  const response = await fetch(url);
  const destination = await response.json();
  return destination;
};

const fetchRides = async (destination) => {
  const url = `http://localhost:3000/api/rides/get/${destination}`;
  const body = {
    method: 'POST', 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = await response.json();
  return rides;
};

const fetchPickups = async (location) => {
  const url = `http://localhost:3000/api/pickup/get/${location}`;
  const body = {
    method: "POST", 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const pickups = await response.json();
  return pickups;
};

const fetchGeocode = async (address) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${geoKey}`;  
  try {
    const response = await fetch(url);
    const locationData = await response.json();
    if (!locationData) {
      return new Error('error')
    }
    return locationData;
  } catch (error) {
    throw error;
  }
}; 

const submitNewRide = async (ride) => {
  const url = `http://localhost:3000/api/rides/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(ride),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const rideId = await response.json();
  return rideId;
};

const submitNewPickup = async (location) => {
  const url = `http://localhost:3000/api/pickup/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(location),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const pickupId = await response.json();
  return pickupId;
};

const fetchRidesPassengers = async (locationId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/passengers/${locationId}`;
  const body = {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const passengers = await response.json();
  return passengers;
};

const postRidesPassengers = async (newPassenger) => {
  const url = `http://localhost:3000/api/rides_passengers/new`;   
  const body = {
    method: "POST",
    body: JSON.stringify(newPassenger),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const ridePassenger = await response.json();
  return ridePassenger;
};

const postNewProfile = async (profile) => {
  const url = `http://localhost:3000/api/profiles/new`;
  const body = {
    method: "POST",
    body: JSON.stringify(profile),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const profileInfo = await response.json();
  return profileInfo;
};

const removePassengerRide = async (rideId, passengerId, destinationId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/passengers/${passengerId}/destination/${destinationId}`; 
  const body = {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const deletedId = await response.json();
  return deletedId;
};

const fetchUserRides = async (userId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/rides/${userId}`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = response.json();
  return rides;
};

const fetchRidesFromUser = async (rideId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/get/user/`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const ridesInfo = await response.json();
  return ridesInfo;
};

const reverseGeoCode = async (lat, lng) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${geoKey}`;
  const response = await fetch(url);
  const geoData = await response.json();
  return geoData;
}

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewRide,
  submitNewPickup,
  fetchRidesPassengers,
  postRidesPassengers,
  postNewProfile,
  removePassengerRide,
  fetchUserRides,
  fetchRidesFromUser,
  reverseGeoCode
};