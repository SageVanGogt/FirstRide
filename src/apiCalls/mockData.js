export const mockUser = {
  id:1,
  user_id:"456",
  user_img:"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
  bio:"I like to eat",
  rating:"Solid 7"
}