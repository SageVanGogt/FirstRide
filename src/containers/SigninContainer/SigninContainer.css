
.signin-container {
  display: flex;
  justify-content: center;
  align-items: center;  
  width: 50%;
}

.signin-form {
  display: flex;
  flex-direction: column;
  width: 70%;  
}

.signin-input {
  height: 40px;
  margin-top: 20px;
}