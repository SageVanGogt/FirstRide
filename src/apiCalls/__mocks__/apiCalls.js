import * as MOCK from './../mockData';

const signinUser = jest.fn().mockImplementation(() => 
  Promise.resolve({
    user: [{
    "id":1,
    "user_id":"456",
    "user_img":"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
    "bio":"I like to eat",
    "rating":"Solid 7"
    }]
  })
);

const signupUser = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1
}));

const fetchDestination = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1,
  "location_name": "Red Rocks",
  "location_lat_lng": "12341, 123412"
}));

const fetchRides = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockRides));

const fetchPickups = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockPickups));

const fetchGeocode = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockGeoInfo));

const submitNewRide = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const submitNewPickup = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const postRidesPassengers = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewPickup,
  submitNewRide,
  postRidesPassengers
}