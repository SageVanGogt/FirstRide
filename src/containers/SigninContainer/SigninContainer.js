import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signinUser } from './../../apiCalls/apiCalls';
import * as actions from './../../actions/user';
import './SigninContainer.css';

export class SigninContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const response = await signinUser(this.state);
    const user = {
      id: response.user[0].id,
      user_name: response.user[0].user_name
    }
    this.props.signinUser(user);
  }

  render() {
    return (
      <div className="signin-container">
        <form 
          action="submit" 
          onSubmit={this.handleSubmit}
          className="signin-form">
          <input 
            className="signin-input"
            type="text" 
            name="email"
            onChange={this.handleChange}
            placeholder="email"
          />
          <input 
            className="signin-input"
            type="password" 
            name="password"
            onChange={this.handleChange}
            placeholder="password"
          />
          <input 
            className="cred-submit-btn"
            type="submit"
            value="Login"            
          />
        </form>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({

})

export const mapDispatchToProps = (dispatch) => ({
  signinUser: (userData) => dispatch(actions.signinUser(userData))
})

export default connect(mapStateToProps, mapDispatchToProps)(SigninContainer);