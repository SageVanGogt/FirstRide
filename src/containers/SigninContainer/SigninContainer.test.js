import React from 'react';
import { shallow } from 'enzyme';
import { SigninContainer, mapDispatchToProps } from './SigninContainer';
import { signinUser } from './../../apiCalls/apiCalls';
import * as MOCK from './../../apiCalls/mockData';

jest.mock('./../../apiCalls/apiCalls');

describe('SigninContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SigninContainer />)
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleChange', () => {
    it('should update the state when input value changes', () => {
      const mockEvent = {
        target: {
          value: 'thurmanvogt@gmail.com',
          name: "email"
      }}
      const expected = {
        email: 'thurmanvogt@gmail.com',
        password: ''
      }
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('handleSubmit', () => {
    it('should call signinUser with the correct params', async () => {
      const expected = {
        email: "thurmanvogt@gmail.com",
        password: "sage"
      }
      wrapper.setState(expected);
      Promise.resolve(wrapper.instance().handleSubmit());
      
      expect(signinUser).toHaveBeenCalledWith(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should be called with the correct params', () => {
      let mockDispatch = jest.fn();
      let mappedProps = mapDispatchToProps(mockDispatch);
      let mockArg = MOCK.mockUser;
      let expected = {
        type: "SIGNIN_USER",
        user: MOCK.mockUser
      }
      mappedProps.signinUser(mockArg);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    })
  })
})