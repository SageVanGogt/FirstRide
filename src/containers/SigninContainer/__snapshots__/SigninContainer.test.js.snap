// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SigninContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          action="submit"
          onSubmit={[Function]}
>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="text"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <input
                    type="submit"
          />
</form>,
        <Connect(SignupContainer) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "submit",
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <input
              type="submit"
/>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            action="submit"
            onSubmit={[Function]}
>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="text"
                        value=""
            />
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <input
                        type="submit"
            />
</form>,
          <Connect(SignupContainer) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "submit",
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
/>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <input
                type="submit"
/>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
